apps/src/applab/applab.js

```diff
@@ -1315,6 +1326,21 @@ Applab.execute = function() {
     }
   }
 
+  if (Applab.makerlabController) {
+    Applab.makerlabController.ensureBoardConnected(
+        function (e) {
+          console.log("Error connecting to board.", e);
+        },
+        function () {
+          Applab.makerlabController.installComponentsOnInterpreter(codegen, Applab.JSInterpreter);
+          Applab.beginVisualizationRun();
+        });
```

Comment:
It would be nice to keep as much of this logic in markerlabController as possible. Maybe we do something like:
```
Applab.makelabController.beginExecution(Applab.beginVisualizationRun)
```
Then let `beginExecution` handle calling ensureBoardConnected, logging errors, installingComponents to the interpreter, and then finally calling it's callback. Thoughts?

Also, thoughts on using a promise instead of a callbac? Such that we would do:
```
Applab.makerlabController.beginExecution().then(beginVisualizationRun)
```